src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -180,8 +184,9 @@ private async Task WriteBatchBytesAsync(PendingRender pending)
                     return;
                 }
 
-                Log.BeginUpdateDisplayAsync(_logger, _client.ConnectionId, pending.BatchId, pending.Data.Length);
-                await _client.SendAsync("JS.RenderBatch", Id, pending.BatchId, pending.Data);
+                Log.BeginUpdateDisplayAsync(_logger, _client.ConnectionId, pending.BatchId, pending.Data.Count);
+                var segment = new ArraySegment<byte>(pending.Data.Buffer, 0, pending.Data.Count);
+                await _client.SendAsync("JS.RenderBatch", Id, pending.BatchId, segment);
```

Comment:
It would be ideal if we could dispose the `memoryStream.ArrayBuilder` right here after the async send. Do you happen to know whether SignalR completes this task once the send has been issued, or does it complete the task immediately and continue to hold a reference to the data while the message is on a queue for dispatch?

Reply:
We cannot dispose these buffers until the client ACKs the receipt. 
cc @BrennanConroy for the confirmation, but `SendAsync` does not guarantee delivery to the client, in fact it does not throw if the send failed.

Reply:
I don't know this code at all, but the buffer should no longer be in use by SignalR after the `await` if this is equivalent to `Context.SendAsync(...)` in a hub method.

Reply:
Then it sounds like we *should* be able to dispose at this point. If not I’d be very interested to know how the data could still be used.

Reply:
@pranavkm Let me know when this PR is ready for re-review. I'm still hoping we might be able to release the array at this point, without waiting for the client's ACK, because that would make it much harder to hold up server memory by failing to send back timely ACKs.

Reply:
We discussed this offline. We need these buffers around to support client disconnects. Leaving this as-is for now. We have follow up items to limit queuing these up.